server,pgwire: make more log message bits non-redactable

knz opened the pull request:
Commit prefix from  #53764
see individual commits for details

Release justification: low risk, high benefit changes to existing functionality

knz committed:
pgwire: expose `root` and `node` client login names as non-redactable

Release note: None

Release justification: low risk, high benefit changes to existing functionality

knz committed:
kvserver: expose gossip functions as non-redactable

Release note: None

Release justification: low risk, high benefit changes to existing functionality

knz committed:
roachpb,server: make store initialization messages non-redactable

Release note: None

Release justification: low risk, high benefit changes to existing functionality

knz committed:
server: make more server start-up messages non-redactable

Release note: None

Release justification: low risk, high benefit changes to existing functionality

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53312)
<!-- Reviewable:end -->

knz committed:
roachpb: make version objects safe for reporting

Release note: None

Release justification: low risk, high benefit changes to existing functionality

knz committed:
jobs: make a duration safe for reporting

Release note: None

Release justification: low risk, high benefit changes to existing functionality

knz committed:
sql: make schema IDs safe for reporting

Release note: None

Release justification: low risk, high benefit changes to existing functionality

knz committed:
jobs: make job types safe for reporting

Release note: None

Release justification: low risk, high benefit changes to existing functionality

knz committed:
server: make drain details non-redactable

Release note: None

Release justification: low risk, high benefit changes to existing functionality

knz commented:
cc @spaskob @lucy-zhang to review SQL schema and jobs adjustments

knz commented:
This work should be improved once https://github.com/cockroachdb/redact/issues/2 is solved.

spaskob reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53312#-:-MFf5wIS5WerBFdWMWfH:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @irfansharif)



<!-- Sent from Reviewable.io -->

irfansharif commented on pkg/cli/start.go:
```diff
@@ -607,9 +608,10 @@ If problems persist, please see %s.`
-			fmt.Fprintf(tw, "CockroachDB node starting at %s (took %0.1fs)\n", timeutil.Now(), timeutil.Since(tBegin).Seconds())
-			fmt.Fprintf(tw, "build:\t%s %s @ %s (%s)\n", info.Distribution, info.Tag, info.Time, info.GoVersion)
-			fmt.Fprintf(tw, "webui:\t%s\n", serverCfg.AdminURL())
+			redact.Fprintf(tw, "CockroachDB node starting at %s (took %0.1fs)\n", timeutil.Now(), timeutil.Since(tBegin).Seconds())
```
I'm assuming we're going to do something here to make this less of a footgun? (was discussed yesterday in the team meeting).

knz replied:
Yes once https://github.com/cockroachdb/redact/pull/4 is merged I will rebase on top of that and remove the footgun here.

knz replied:
Done.

irfansharif reviewed (approved):
LGTM, hopefully this was relaxing work.

lucy-zhang commented on docs/generated/redact_safe.md:
```diff
@@ -10,6 +10,9 @@ pkg/roachpb/metadata.go | `RangeID`
 pkg/roachpb/metadata.go | `ReplicaID`
 pkg/roachpb/metadata.go | `RangeGeneration`
 pkg/roachpb/metadata.go | `ReplicaType`
+pkg/sql/catalog/descpb/structured.go | `ID`
```
Thanks. Is it possible to also add some of the other descriptor IDs in `descpb/structured.go` (`ColumnID`, `IndexID`, `FamilyID`, `MutationID`)?

knz replied:
taking note, will do

lucy-zhang replied:
Thanks. I think we want `descpb.DescriptorVersion` as well.

knz committed:
deps: bump cockroachdb/errors and cockroachdb/redact

This picks up improved support for `%T` in format strings,
as well as better support for `%w`.

Release justification: low risk, high benefit changes to existing functionality

Release note: None

knz committed:
colexecerror: require an `error` argument for InternalError()

Prior to this patch, `colexecerror.InternalError()` would take any
type as input. This was encouraging callers to pass not only constant
strings and errors, it was also encouraging uses of `fmt.Sprintf`

However, given that these errors ultimately make their way to crash
reporting, we really want as much details as possible in crash
reports. Simple strings don't cut it and are considered fully unsafe
for reporting.

What we want instead if a discrete separation of safe and unsafe
details.  The machinery to do this is available in the `redact`
package, but for ease of use the `errors` library also does it.

So this patch changes `InternalError()` to require an `error`
argument, and invokes `errors.AssertionFailedf` or `log.PanicAsError`
in the call points.

This API can be further simplified in the future by making
`InternalError`
responsible for constructing the error object, for example

```go
InternalErrorf(format string, args ...interface{}) {
  panic(errors.AssertionFailedf(format, args...))
}
```

This improvement is left for a later change though.

Release justification: low risk, high benefit changes to existing functionality

Release note: None

knz committed:
*: convert more panic strings to error objects

Error objects in panics allow us to expose more safe details for
telemetry reporting.

Release justification: low risk, high benefit changes to existing functionality

Release note: None

knz committed:
descpb: opt more simple integer types into safe reporting

Release justification: low risk, high benefit changes to existing functionality

Release note: None

knz reviewed (commented):
> Thanks. I think we want descpb.DescriptorVersion as well.

Done.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53312#-:-MGDSb9rAi8hpjihaO9p:b-8z8fsz)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

knz commented:
tfyrs

bors r=spaskob,irfansharif
